### PR TITLE
When node is in upgrade state, allow only upgrade specific recipes in runlist

### PR DESCRIPTION
**Pull request by jsuchome**
Back ported from https://github.com/crowbar/crowbar-core/pull/179 cc @rhafer

### CODE REVIEW

File: crowbar_framework/app/models/node_object.rb [hunk 1]
```diff
@@ -659,7 +659,12 @@ def rebuild_run_list
     crowbar["run_list_map"] = {} if crowbar["run_list_map"].nil?
 
     # Cull by state
-    map = crowbar["run_list_map"].select { |k,v| v["states"].include?("all") or v["states"].include?(self.crowbar['state']) }
+    map = crowbar["run_list_map"].select do |k, v|
+      v["states"].include?(state) ||
```
Comment (vuntz) [inline crowbar_framework/app/models/node_object.rb]: Actually, I think the move from `self.crowbar['state']` to `state` is wrong: we should never do something depending on `noupdate`, imho.

File: crowbar_framework/app/models/node_object.rb [hunk 2]
```diff
@@ -659,7 +659,12 @@ def rebuild_run_list
...
+      # When node is in upgrade state, allow only upgrade specific recipes
+      (v["states"].include?("all") && !state.include?("upgrade"))
```
Comment (vuntz) [inline crowbar_framework/app/models/node_object.rb]: Isn't the state named crowbar_upgrade?

Comment (jsuchome) [inline crowbar_framework/app/models/node_object.rb]: Yes, that's just for potential other `_upgrade` state. But better to leave the only name we currently have, I guess...

**Comment by rhafer** [comment]
+1

**Comment by vuntz** [comment]
+1

**Pull request merged**